### ISSUE TITLE
Unable to reproduce `SuperCLEVR`

### ISSUE THREAD

**Issue by lerogo** [Deep-Agent/R1-V#170]
8GPUs, 200 steps, evaluate checkpoint-100

**The results we reproduced differ significantly from the reported results in the repo and blog.**
**The reproduction result is:**
**71.50%**

[Qwen2-VL-2B-Instruct_GRPO_CLEVR_200STEP_20250320143823.json](https://github.com/user-attachments/files/19362199/Qwen2-VL-2B-Instruct_GRPO_CLEVR_200STEP_20250320143823.json)

training log:  [training_log_SuperCLEVR.log](https://github.com/user-attachments/files/19362188/training_log_SuperCLEVR.log)

training script:
```bash
timestamp=$1
echo "timestamp: ${timestamp}"

DATA_PATH=leonardPKU/clevr_cogen_a_train
CKPT_PATH=Qwen2-VL-2B-Instruct

SAVE_PATH="./output/Qwen2-VL-2B-Instruct_GRPO_CLEVR_200STEP_${timestamp}"

export DEBUG_MODE="true" # Enable Debug if you want to see the rollout of model during RL
export LOG_PATH="./${SAVE_PATH}/debug_log.txt"

mkdir -p ${SAVE_PATH}

torchrun --master_addr ${MASTER_ADDR} --master-port ${MASTER_PORT} \
    --nnodes ${WORLD_SIZE} --node_rank ${RANK} --nproc-per-node=${GPUS} \
    src/r1-v/src/open_r1/grpo.py \
    --output_dir ${SAVE_PATH} \
    --model_name_or_path ${CKPT_PATH} \
    --dataset_name ${DATA_PATH} \
    --deepspeed src/r1-v/local_scripts/zero3.json \
    --max_prompt_length 1024 \
    --max_completion_length 1024 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 2 \
    --logging_steps 1 \
    --bf16 \
    --report_to tensorboard \
    --gradient_checkpointing false \
    --attn_implementation flash_attention_2 \
    --max_pixels 401408 \
    --num_train_epochs 1 \
    --max_steps 200 \
    --run_name "Qwen2-VL-2B-GRPO-CLEVR-70k_200STEP_${timestamp}" \
    --save_steps 50 \
    --save_only_model true \
    --num_generations 8 \
    2>&1 | tee -a "./${SAVE_PATH}/training_log.log"
```

curve:
<img width="1335" alt="Image" src="https://github.com/user-attachments/assets/a00eb998-cb1f-416f-9896-3cc5e5041dfe" />

**Comment by lerogo** [comment]
@chenllliang 

Did you find any problems? Why is the gap so big?

**Comment by wuhuikai** [comment]
I think that it's caused by `max_steps=200`, which makes the learning rate drop faster than the default script. The original script doesn't contain `max_steps`. It only contains `num_train_epochs=2`, which implies that `max_steps=num_train_epochs*len(dataset)//batch_size`. This value is much larger than `200`.

**Comment by lerogo** [comment]
> I think that it's caused by `max_steps=200`, which makes the learning rate drop faster than the default script. The original script doesn't contain `max_steps`. It only contains `num_train_epochs=2`, which implies that `max_steps=num_train_epochs*len(dataset)//batch_size`. This value is much larger than `200`.

Thank you for your reply.

The training steps afterwards are meaningless, should the learning rate be adjusted manually?
Also, another question: the blog states a learning rate of 1e-6, but the script does not specify by default, and the default for transformer is ~~5e-5~~ 1e-6(grpo).
What exactly is the learning rate, and does it need to be explicitly specified?

**Comment by lerogo** [comment]
> I think that it's caused by `max_steps=200`, which makes the learning rate drop faster than the default script. The original script doesn't contain `max_steps`. It only contains `num_train_epochs=2`, which implies that `max_steps=num_train_epochs*len(dataset)//batch_size`. This value is much larger than `200`.

We will try it again. 
Set the learning rate to 1e-6, and the epoch is set to 2.

**Comment by lerogo** [comment]
**The new reproduction result is: 79.00%**
There is still a big gap.

**Comment by lerogo** [comment]
Can you show the curve images including loss and learning rate?
The picture below is the new experiment, the result of 100 steps is **79.00%**.

<img width="1323" alt="Image" src="https://github.com/user-attachments/assets/3ba41237-56c5-4cff-b74d-ab8cc1057600" />

**Comment by lerogo** [comment]
> I think that it's caused by `max_steps=200`, which makes the learning rate drop faster than the default script. The original script doesn't contain `max_steps`. It only contains `num_train_epochs=2`, which implies that `max_steps=num_train_epochs*len(dataset)//batch_size`. This value is much larger than `200`.

@chenllliang @yuezih 
We need the complete training script, parameter configuration, and training logs to check the issues we reproduced.

**Comment by lerogo** [comment]
@chenllliang @yuezih
